### PR TITLE
[REVIEW][skip-ci] updated docs

### LINKED ISSUE #848: [DOC] Update main docs
The README should include a section on Graph sizes to required GPU memeory

The Source build should:
* discuss clang-format
* highlight use of build.sh first

### PR DESCRIPTION
**Pull request by BradReesWork**
updated main docs

closes #848

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: README.md [hunk 1]
```diff
@@ -86,6 +86,15 @@ cuGraph provides an auto-renumbering feature, enabled by default, during Graph c
 
 cuGraph is constantly being updatred and improved. Please see the [Transition Guide](TRANSITIONGUIDE.md) if errors are encountered with newer versions
 
+## Graph Sizes and GPU Memory Size
+As a simple rule of thumb, the amount of GPU memory should be about twice the size of the data size.  That gives overhead for the CSV reader and other transform functions.  There are ways around the rule but using smaller data chunks.
```
Comment (afender) [inline README.md:90]: I'd recommend adding that "This estimate varies depending on the algorithm and the connectivity pattern of the graph."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ cuGraph, and all of RAPIDS in general, is an open-source project where we encour
 If you are ready to contribute, jump right to the [Contribute Code](#code) section.
 
 
+__Style Formating Tools:__
```
Comment (seunghwak) [inline CONTRIBUTING.md:11]: should we better move this to the "Style Guide" section?

https://github.com/rapidsai/cugraph/pull/859/files#diff-6a3371457528722a734f3c51d9238c13R151

And
### Style Guild
=>
### Style Guide

File: README.md [hunk 3]
```diff
@@ -86,6 +86,15 @@ cuGraph provides an auto-renumbering feature, enabled by default, during Graph c
 
 cuGraph is constantly being updatred and improved. Please see the [Transition Guide](TRANSITIONGUIDE.md) if errors are encountered with newer versions
 
+## Graph Sizes and GPU Memory Size
```
Comment (seunghwak) [inline README.md:89]: (Irrelevant to the line above).

In
https://github.com/rapidsai/cugraph/pull/859/files#diff-04c6e90faac2675aa89e2176d2eec7d8R5

For users familar with C/CUDA and graph structures, a C++ API is also provided.  However, there is less type and structure checking at the C layer.
=>

For users famil"i"ar with C"++"/CUDA and graph structures, a C++ API is also provided.  However, there is less type and structure checking at the C"++" layer.

File: SOURCEBUILD.md [hunk 4]
```diff
@@ -4,27 +4,33 @@ The following instructions are for users wishing to build cuGraph from source co
...
-CUDA requirement:
+__Compiler__:
+* `gcc`         version 5.4+
+* `nvcc`        version 10.0+
+* `cmake`       version 3.12
 
+__CUDA:__
 * CUDA 10.0+
 * NVIDIA driver 396.44+
 * Pascal architecture or better
```
Comment (seunghwak) [inline SOURCEBUILD.md:18]: No need to add clang-format and flake8 here?

https://github.com/rapidsai/cugraph/pull/859/files#diff-6a3371457528722a734f3c51d9238c13R12

File: SOURCEBUILD.md [hunk 5]
```diff
@@ -74,9 +80,50 @@ conda env update --name cugraph_dev --file conda/environments/cugraph_dev_cuda10
...
+
+# make parallelism options can also be defined: Example build jobs to 4 (make -j4)
+$ PARALLEL_LEVEL=4 ./build.sh libcugraph
+
+Note that the libraries will be installed to the location set in `$PREFIX` if set (i.e. `export PREFIX=/install/path`), otherwise to `$CONDA_PREFIX`.
+```
+
+
+## Building each section independently
+#### Build and Install the C/CUDA `libcugraph` Library
```
Comment (seunghwak) [inline SOURCEBUILD.md:123]:
```suggestion
#### Build and Install the C++/CUDA `libcugraph` Library
```
Need to search for C/CUDA and replace C with C++

File: CONTRIBUTING.md (hunk 2 above)
Comment (BradReesWork) [reply CONTRIBUTING.md:11]: Should have pointers to the style guides

File: README.md (hunk 3 above)
Comment (BradReesWork) [reply README.md:89]: fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (BradReesWork) [reply CONTRIBUTING.md:11]: added link

File: SOURCEBUILD.md (hunk 4 above)
Comment (BradReesWork) [reply SOURCEBUILD.md:18]: Nope - this is just how to compile the code / not modify it.

File: SOURCEBUILD.md (hunk 5 above)
Comment (BradReesWork) [reply SOURCEBUILD.md:123]: done

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -147,6 +151,7 @@ implementation of the issue, ask them in the issue instead of the PR.
 ### Style Guild
```
Comment (seunghwak) [inline CONTRIBUTING.md:151]:
```suggestion
### Style Guide
```

File: README.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
...
+The [RAPIDS](https://rapids.ai) cuGraph library is a collection of GPU accelerated graph algorithms that process data found in [GPU DataFrames](https://github.com/rapidsai/cudf).  The vision of cuGraph is _to make graph analysis ubiquitous to the point that users just think in terms of analysis and not technologies or frameworks_.  To realize that vision, cuGraph operators, at the Python layer, on GPU DataFrames, allowing for seamless passing of data between ETL tasks in [cuDF](https://github.com/rapidsai/cudf) and machine learning tasks in [cuML](https://github.com/rapidsai/cuml).  Data scientist familiar with Python will quickly pick up how cuGraph integrates with the Pandas-like API of cuDF.  Likewise, user familiar with NetworkX will quickly reconnize the NetworkX-like API provided in cuGraph, with the goal being to allow existing code to be ported with minimal effort into RAPIDS.  For users familiar with C++/CUDA and graph structures, a C++ API is also provided.  However, there is less type and structure checking at the C layer.
```
Comment (seunghwak) [inline README.md:5]:
```suggestion
The [RAPIDS](https://rapids.ai) cuGraph library is a collection of GPU accelerated graph algorithms that process data found in [GPU DataFrames](https://github.com/rapidsai/cudf).  The vision of cuGraph is _to make graph analysis ubiquitous to the point that users just think in terms of analysis and not technologies or frameworks_.  To realize that vision, cuGraph operators, at the Python layer, on GPU DataFrames, allowing for seamless passing of data between ETL tasks in [cuDF](https://github.com/rapidsai/cudf) and machine learning tasks in [cuML](https://github.com/rapidsai/cuml).  Data scientist familiar with Python will quickly pick up how cuGraph integrates with the Pandas-like API of cuDF.  Likewise, user familiar with NetworkX will quickly reconnize the NetworkX-like API provided in cuGraph, with the goal being to allow existing code to be ported with minimal effort into RAPIDS.  For users familiar with C++/CUDA and graph structures, a C++ API is also provided.  However, there is less type and structure checking at the C++ layer.
```

**Pull request merged**